contracts/CodexTitleCore.sol

```diff
@@ -55,11 +55,12 @@ contract CodexTitleCore is CodexTitleMetadata, CodexTitleFees {
     // Add a new token to the allTokens array
     super._mint(_to, newTokenId);
 
-    // Add new token data to the newly created token
-    // Note that we aren't using the struct here so we can directly write to the imageHashes dynamic storage array
-    tokenData[newTokenId].nameHash = _nameHash;
-    tokenData[newTokenId].descriptionHash = _descriptionHash;
-    tokenData[newTokenId].imageHashes.push(_imageHash);
+    // Add metadata to the newly created token
+    tokenData[newTokenId] = CodexTitleData({
```

Comment:
@johnhforrest is there any problem using the struct like this? Seems like you were specifically trying to avoid this, but I think this is more efficient than looping over `_fileHashes` and pushing the hashes onto `tokenData[newTokenId].fileHashes`

Reply:
This was required when we were appending to the array, but since it's being replaced now that syntax is not required.

Can you leave a TODO here to make sure we evaluate the gas costs? I have a feeling recreating the array every time is going to cost significantly more.